### PR TITLE
Feature/notify method

### PR DESCRIPTION
**Pull request by 89luca89**
Hi all,

As discussed in the mailing list here: https://lists.opensuse.org/archives/list/kubic@lists.opensuse.org/thread/E6ZYOXFKRM5BODY25PEWZRBCMEFYS66W/

This pull request aims to add a notify method (`reboot_via_notify`) so that on MicroOS Desktop setups we can let the user know that the t-u timer applied updates successfully.
This should be a more suitable way to apply updates automatically on desktop setups where 3:30am reboots are not advisable.

This PR also adds `Persistent=true` to the systemd-timer, in order to ensure that updates are performed if the machine is in sleep/shutdown during the timer window.

This PR goes in hand with this other tool: https://github.com/89luca89/transactional-update-notifier
Which is the client/server app that is called by the `reboot_via_notify` method, in order to sent notifications to all logged user graphically.

To summarize this PR implements:

- a new `reboot_via_notify` method in the t-u script
- the correspondent reboot method to `Reboot.cpp`
- persistence to the systemd-timer

Also the notifier tool supports reporting faliures, we can evaluate if using this feature or not in the t-u script

Signed-off-by: Luca Di Maio <luca.dimaio1@gmail.com>

### CODE REVIEW

**Review by sysrich — APPROVED** [review]
Yes yes a thousand times yes :)

**Comment by 89luca89** [comment]
> Yes yes a thousand times yes :)

Thanks :)

This will require on MicroOS Desktop a specific settings for the reboot method (set to notify)
Probably could be provided as a separate package required by the desktop patterns? Not sure how we can provide this I'm not a packaging expert :joy:

**Comment by laenion** [comment]
Sorry for the delay, I'm currently on vacation. Will have a look at it on Tuesday when I'm back, but I can say already that I like the general idea a lot!

**Comment by laenion** [comment]
Released transactional-update 4.1.0 with this PR included and submitted to Factory: https://build.opensuse.org/request/show/1031369

Thanks a lot for your work, I'm really looking forward to use it!

**Comment by 89luca89** [comment]
> Released transactional-update 4.1.0 with this PR included and submitted to Factory: https://build.opensuse.org/request/show/1031369
> 
> Thanks a lot for your work, I'm really looking forward to use it!

Thanks a lot for merging!
